pkg/lib/cockpit/channel.ts

```diff
@@ -0,0 +1,287 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import type { JsonObject } from './_internal/common';
+import { Transport, ensure_transport, transport_globals } from './_internal/transport';
+import { EventEmitter } from './event';
+
+export type ChannelPayload = string | Uint8Array;
+
+export interface BaseChannelOptions extends JsonObject {
+    command?: never;
+    channel?: never;
+    binary?: boolean;
+    host?: string;
+    payload?: string;
+    superuser?: "try" | "require";
```

Comment:
So, no boolean at all any more? OK for now, but we also have at least one case of
```
superuser: this.props.superuser,
```
This is rather sloppy, and needs to be typed by itself; but perhaps accepting a `boolean` wouldn't hurt?

Reply:
Sure.  I don't have a strong opinion on this.  I think @jelly might, though.

Reply:
No strong opinion from me either. We can also ignore this for now and relax it if/when we need it. That's probably the better approach anyway -- I don't think we'll need that class in c-podman etc. anyway, that all feels like "inside of cockpit.git".

Reply:
Resolving -- let's keep this for now, easier to extend than to reduce.

Reply:
We collided -- but still, you can write optional dict arguments in Python with some `**` syntax. It's ugly, but I don't think we'll need that often.